### PR TITLE
feat(cdc): allow updating of failed changefeed

### LINKED ISSUE #8018: Support editing the configuration of the changefeed in failed status
### Is your feature request related to a problem?

when the configuration is wrong , the changefeed will fail. But the user can't edit this changefeed and resume it .

### Describe the feature you'd like

The user can edit the configuration when the status of the changefeed is failed or paused.

### Describe alternatives you've considered

_No response_

### Teachability, Documentation, Adoption, Migration Strategy

_No response_

### LINKED ISSUE #8478: TiCDC  should support update the  configuration of a changefeed in a failed state
### Is your feature request related to a problem?

if a changefeed failed due to the wrong configuration , the customer can't update it .

### Describe the feature you'd like

as title described.

### Describe alternatives you've considered

_No response_

### Teachability, Documentation, Adoption, Migration Strategy

_No response_

### PR DESCRIPTION
**Pull request by charleszheng44**
### What problem does this PR solve?
Issue Number: close #8478 
Issue Number: close #8018 

### What is changed and how it works?
Allow updating of the failed changefeed

#### Tests 

 - [x] Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

##### Will it cause performance regression or break compatibility?
No

##### Do you need to update user documentation, design documentation or monitoring documentation?
No

### Release note 
```release-note
`None`
```

### CODE REVIEW

**Review by asddongmen — APPROVED** [review]

File: cdc/api/v1/api.go [hunk 1]
```diff
@@ -441,10 +441,18 @@ func (h *OpenAPI) UpdateChangefeed(c *gin.Context) {
...
 	}
-	if info.State != model.StateStopped {
-		_ = c.Error(cerror.ErrChangefeedUpdateRefused.GenWithStackByArgs("can only update changefeed config when it is stopped"))
+
+	switch info.State {
+	case model.StateFailed, model.StateStopped:
+	default:
+		_ = c.Error(
+			cerror.ErrChangefeedUpdateRefused.GenWithStackByArgs(
+				"can only update changefeed config when it is stopped",
```
Comment (asddongmen) [inline cdc/api/v1/api.go:450]: or failed?

File: cdc/api/v2/changefeed.go [hunk 2]
```diff
@@ -329,11 +329,18 @@ func (h *OpenAPIV2) updateChangefeed(c *gin.Context) {
...
-	if oldCfInfo.State != model.StateStopped {
-		_ = c.Error(cerror.ErrChangefeedUpdateRefused.
-			GenWithStackByArgs("can only update changefeed config when it is stopped"))
+
+	switch oldCfInfo.State {
+	case model.StateStopped, model.StateFailed:
+	default:
+		_ = c.Error(
+			cerror.ErrChangefeedUpdateRefused.GenWithStackByArgs(
+				"can only update changefeed config when it is stopped",
```
Comment (asddongmen) [inline cdc/api/v2/changefeed.go:338]: ditto

**Review by nongfushanquan — APPROVED** [review]

File: cdc/api/v1/api.go (hunk 1 above)
Comment (charleszheng44) [reply cdc/api/v1/api.go]: fixed

File: cdc/api/v2/changefeed.go (hunk 2 above)
Comment (charleszheng44) [reply cdc/api/v2/changefeed.go]: fixed

**Pull request merged**